lib/editor/components/ScheduleExceptionForm.js

```diff
@@ -197,18 +216,26 @@ export default class ScheduleExceptionForm extends Component<Props> {
             <ControlLabel>
               <small>On these dates*</small>
             </ControlLabel>
-            {dates && dates.length
-              ? dates.map((date, index) => (
-                <ExceptionDate
-                  index={index}
-                  key={index}
-                  date={date}
-                  // Exception dates expects a boolean
-                  isNotValid={Boolean(this._checkValidation(`dates-${index}`))}
-                  {...this.props}
-                />
-              ))
-              : <div>No dates specified</div>
+            {ranges && ranges.map((range, index) => (
+              <ExceptionDateRange
+                index={index}
+                key={index}
+                range={range}
+                isInvalid={this._checkRangeValidation(dates.findIndex(date => date === range.startDate), dates.findIndex(date => date === range.endDate))}
+                {...this.props}
+              />
+            ))}
+            {parsedDates && parsedDates.map((date, index) => (
+              <ExceptionDate
+                index={dates.findIndex(d => d === date)}
+                key={index}
+                date={date}
+                isInvalid={Boolean(this._checkValidation(`dates-${dates.findIndex(d => d === date)}`))}
```

Comment:
We can extract `dates.findIndex(d => d === date)`

Reply:
updated in https://github.com/ibi-group/datatools-ui/pull/828/commits/b8ea974dc8ba296b92d2b964760b893be231af36